### PR TITLE
Reduce the number of github requests in the backport bot

### PR DESCRIPTION
**Pull request by Therzok**
Do a quick assessment of the input before doing any API calls.

Tested some inputs offline:
```pwsh
$text = "@vsmacbot backport release-17.0/devel"

if ($text -match "@(?<botName>.+) backport (?<branchName>.+)") {
 $Matches
}
```

```
Name                           Value
----                           -----
botName                        vsmacbot
branchName                     release-17.0/devel
0                              @vsmacbot backport release-17.0/devel
```

### CODE REVIEW

**Comment by Therzok** [comment]
@cadsit @mjbond-msft ping?

File: action.yml [hunk 1]
```diff
@@ -43,6 +43,12 @@ runs:
         $statusCode = 0
         $message = ""
         try {
+          Write-Host "Parsing ${{ inputs.comment_body }}"
+          if ("${{ inputs.comment_body }}" -match "@(?<botName>.+) backport (?<branchName>.+)") {
+            $botName = $Matches.botName;
+            $backportTargetBranch = $Matches.branchName;
+          }
```
Comment (cadsit) [inline action.yml:50]: Nit: I really hope we don't have spaces in our branch names, but we might want to make sure we account for those properly. Wanted to make sure the following input is an error mode:
`@vsmacbot backport main please`

```pwsh
if ("${{ inputs.comment_body }}" -match "@(?<botName>.+) backport (?<branchName>(`".+`")|([^\s]+))$") {
  $botName = $Matches.botName;
  $backportTargetBranch = $Matches.branchName.TrimStart("`"").TrimEnd("`"")
} else {
  throw "Could not parse comment body: ${{ inputs.comment_body }}"
}
```

**Review by cadsit — CHANGES_REQUESTED** [review]

File: action.yml (hunk 1 above)
Comment (Therzok) [reply action.yml:50]: Shouldn't the error be thrown only if we find the botname part?

**Review by cadsit — APPROVED** [review]

**Comment by Therzok** [comment]
@cadsit how do I test this commit?

**Comment by Therzok** [comment]
It seems that this isn't the backport action that's used in the vsmac repo.

**Pull request closed without merging**